conformance/packages/conformance-tests/src/name_server/rfc5155.rs

```diff
@@ -0,0 +1,376 @@
+use std::collections::BTreeMap;
+use std::net::Ipv4Addr;
+
+use dns_test::client::{Client, DigSettings, DigStatus};
+use dns_test::name_server::NameServer;
+use dns_test::record::{Record, RecordType, NSEC3};
+use dns_test::{Network, Result, FQDN};
+
+///  An NSEC3 RR is said to "match" a name if the owner name of the NSEC3 RR is the same as the
+///  hashed owner name of that name.
+fn find_match<'a>(name_hash: &str, records: &'a BTreeMap<String, NSEC3>) -> Option<&'a NSEC3> {
```

Comment:
helper code should go at the end of the file and the `#[test]`s should be at the top. it may also make sense to move the `BTreeMap` newtype into the dns-test crate as it may be used in other modules.

Reply:
done